cardano-testnet/src/Testnet/Start/Types.hs

```diff
@@ -99,16 +104,50 @@ instance Default CreateEnvUpdateTime where
   def = CreateEnv
 
 data CreateEnvOptions = CreateEnvOptions
-  { ceoTopologyType :: TopologyType
+  { ceoOnChainParams :: CustomOnChainParams
+  , ceoTopologyType :: TopologyType
   , ceoUpdateTime :: CreateEnvUpdateTime
   } deriving (Eq, Show)
 
 instance Default CreateEnvOptions where
   def = CreateEnvOptions
-    { ceoTopologyType = def
+    { ceoOnChainParams = def
+    , ceoTopologyType = def
     , ceoUpdateTime = def
     }
 
+data CustomOnChainParams
```

Comment:
I would remove `Custom` from the name since they're not all custom. Currently you can have a default or a mainnet set of parameters. `TestnetChainParams` may also be worth considering.